geomesa-fs/geomesa-fs-storage-parquet/src/main/scala/org/locationtech/geomesa/parquet/FilterConverter.scala

```diff
@@ -0,0 +1,273 @@
+                   _: org.opengis.filter.PropertyIsLessThan |
+                   _: org.opengis.filter.PropertyIsLessThanOrEqualTo |
+                   _: org.opengis.filter.PropertyIsGreaterThan |
+                   _: org.opengis.filter.PropertyIsGreaterThanOrEqualTo) =>
+            org.opengis.filter.Filter.INCLUDE
+          case _ => f
+        }
+      case _ => f
+    }
+  }
+
+  protected def dtgFilter(f: org.opengis.filter.Filter): Option[FilterPredicate] = {
+    dtgAttrOpt.map { dtgAttr =>
+      val filters = FilterHelper.extractIntervals(f, dtgAttr).values.map { case (start, end) =>
+        FilterApi.and(
+          FilterApi.gtEq(FilterApi.longColumn(dtgAttr), start.getMillis: java.lang.Long),
+          FilterApi.ltEq(FilterApi.longColumn(dtgAttr), end.getMillis: java.lang.Long)
+        )
+      }
+
+      if (filters.nonEmpty) {
+        Some(filters.reduceLeft(FilterApi.and))
+      } else {
+        None
+      }
+    }
+  }.getOrElse(None)
+
+  protected def geoFilter(f: org.opengis.filter.Filter): Option[FilterPredicate] = {
+    val extracted = extractGeometries(f, geomAttr)
+    if (extracted.isEmpty || extracted.disjoint) {
+      None
+    } else {
+      val xy = extracted.values.map(GeometryUtils.bounds).reduce { (a, b) =>
+        (math.min(a._1, b._1),
+          math.min(a._2, b._2),
+          math.max(a._3, b._3),
+          math.max(a._4, b._4))
+      }
+      Some(
+        List[FilterPredicate](
+          FilterApi.gtEq(FilterApi.doubleColumn("geom.x"), Double.box(xy._1)),
+          FilterApi.gtEq(FilterApi.doubleColumn("geom.y"), Double.box(xy._2)),
+          FilterApi.ltEq(FilterApi.doubleColumn("geom.x"), Double.box(xy._3)),
+          FilterApi.ltEq(FilterApi.doubleColumn("geom.y"), Double.box(xy._4))
+       ).reduce(FilterApi.and)
+      )
+    }
+  }
+
+  protected def attrFilter(gtFilter: org.opengis.filter.Filter): Option[FilterPredicate] = {
+    gtFilter match {
+
+      case and: org.opengis.filter.And =>
+        val res = and.getChildren.flatMap(attrFilter)
+        if (res.nonEmpty) Option(res.reduceLeft(FilterApi.and)) else None
+
+      case or: org.opengis.filter.Or =>
+        val res = or.getChildren.flatMap(attrFilter)
+        if (res.nonEmpty) Option(res.reduceLeft(FilterApi.or)) else None
+
+      case binop: org.opengis.filter.BinaryComparisonOperator =>
+        val name = binop.getExpression1.asInstanceOf[PropertyName].getPropertyName
+        val value = binop.getExpression2.toString
+
+        binop match {
+          case _ if name == geomAttr | dtgAttrOpt.contains(name) =>
+            None
+
+          case _ =>
+            val ad = sft.getDescriptor(name)
+            val binding = ad.getType.getBinding
+            val (objectType: ObjectType, _) = ObjectType.selectType(binding, ad.getUserData)
+            filter(objectType, binop, name, value)
+        }
+
+      case _ =>
+        None
+      // TODO geotools based UDFs?
```

Comment:
What happens if a GT function is applied?  Is it supported or not?  If not, ticket and/or failing unit test.